inference/models/grounding_dino/grounding_dino.py

```diff
@@ -0,0 +1,136 @@
+import os
+import urllib.request
+from time import perf_counter
+
+import torch
+from groundingdino.util.inference import Model
+
+from inference.core.entities.requests.groundingdino import GroundingDINOInferenceRequest
+from inference.core.entities.requests.inference import InferenceRequestImage
+from inference.core.entities.responses.inference import (
+    InferenceResponseImage,
+    ObjectDetectionInferenceResponse,
+    ObjectDetectionPrediction,
+)
+from inference.core.env import MODEL_CACHE_DIR
+from inference.core.models.roboflow import RoboflowCoreModel
+from inference.core.utils.image_utils import load_image_rgb, xyxy_to_xywh
+
+
+class GroundingDINO(RoboflowCoreModel):
+    """GroundingDINO class for zero-shot object detection.
+
+    Attributes:
+        model: The GroundingDINO model.
+    """
+
+    def __init__(
+        self, *args, model_id="groundingdino/groundingdino_swint_ogc", **kwargs
```

Comment:
I'm still spotting some inconsistencies when using the name of grounding dino in string form. Let's change all mentions of grounding dino within a string to `grounding-dino`

Reply:
And any variable name should be `grounding_dino`